projects/fb-openlayers/src/lib/resources/layer-charts.component.ts

```diff
@@ -265,10 +270,16 @@ export class FreeboardChartLayerComponent
                   source: source,
                   preload: 0,
                   zIndex: this.zIndex + parseInt(i),
-                  style: this.applyVectorStyle,
-                  minZoom: minZ,
-                  maxZoom: maxZ
+                  style: isS57 ? this.s57Service.getStyle :   this.applyVectorStyle,
```

Comment:
It would be prefereable to pass 'S-57' as a parameter to `applyVectorStyle()`.

Reply:
This indeed needs to be fixed i think preferably with a  factory  with the chart as input  and a  interface implementing something like  getVectorLayer / getVectorSource.  Do you agree ?

Reply:
I agree.
Should look at using `ol-mapbox-style`.

Reply:
I have refactored the VectorLayer part to a factory now in line with how ol-mapbox-style works